python_ta/checkers/bad_loop_indexing_checker.py

```diff
@@ -0,0 +1,108 @@
+"""checker for bad indexing in a loop.
+"""
+
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from typing import List
+
+
+class BadLoopIndexingChecker(BaseChecker):
+    __implements__ = IAstroidChecker
+    # name is the same as file name but without _checker part
+    name = 'bad_loop_indexing'
+    # use dashes for connecting words in message symbol
+    msgs = {'E9994': ('Bad loop indexing used in this for-loop',
+                      'bad-loop-indexing',
+                      'Used when you have an iteration variable in a for-loop '
+                      'where its only usage is to index the iterable it is iterating over'),
+            }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    def _check_bad_loop_indexing(self, node):
+        """Return whether the usage of the indexing variable in the for-loop is ONLY to index the iterable.
+        True if bad, False if not bad.
+        """
+        # Store the iteration variable
+        index = node.target.name
+
+        # Check if the iterable of the for-loop is of the form "range(len(lst))"
+        iterable_lst = _check_iterable_is_range(node)
+        if iterable_lst:
+            iterable = iterable_lst[0]
+        else:
+            return False
+
+        # Use node_of_class to find all Name nodes.
+        if any(_check_correct_usage(name_node, index, iterable) for name_node in node.nodes_of_class(astroid.Name)):
+            return False
+        # There is at least one 'index' Name node and they are all used badly.
+        elif any(name_node.name == index for name_node in node.nodes_of_class(astroid.Name)):
```

Comment:
Merge this branch with the following one. Note that these two branches have the form

```
if cond:
  return True
else:
  return False
```